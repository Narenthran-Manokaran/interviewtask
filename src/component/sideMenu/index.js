import PropTypes from 'prop-types';
import React, {Component} from 'react';
import { NavigationActions } from 'react-navigation';
import { ScrollView, Text, View } from 'react-native';
import { connect } from 'react-redux';
import { ListItem, Button } from 'react-native-elements';
import RNExitApp from 'react-native-exit-app';
import { fetchData } from './../../actions';
import styles from './sideMenu.style';

const URI = 'https://api.nytimes.com/svc';
const API_KEY = 'api-key=okVdEHSsZSsT6Pluu48AAa35wmjr3BNj';

export class SideMenu extends Component {
  navigateToScreen = (route) => () => {
    const navigateAction = NavigationActions.navigate({
      routeName: route.routeName
    });
    this.props.navigation.dispatch(navigateAction);
    this.props.fetchData(route.params.uri);
  }

  componentDidMount() {
    this.props.fetchData(`${URI}/mostpopular/v2/viewed/1.json?${API_KEY}`);
  }

  render () {
    const { items } = this.props;
    return (
      <View style={styles.container}>
        <ScrollView>
          <View>
            {items.map((item, index) => {
                return (
                  <ListItem
                    key={index}
                    title={item.params.title}
                    onPress={this.navigateToScreen(item)}
                    leftIcon={{ name: 'av-timer' }}
                    rightIcon={{ name: 'angle-right', type:'font-awesome', color:'#517fa4' }}
                  />
                );
            })}
          </View>
        </ScrollView>
        <View style={styles.footerContainer}>
          <Button
            title="Sign Out"
            onPress={() => RNExitApp.exitApp() }
          />
        </View>
      </View>
    );
  }
}

SideMenu.propTypes = {
  navigation: PropTypes.object
};

export const mapStateToProps = (state) => {
    return {
      appData: state.appData
    }
  }
  
export const mapDispatchToProps = (dispatch) => {
  return {
    fetchData: (uri) => dispatch(fetchData(uri))
  }
}

export default connect( mapStateToProps, mapDispatchToProps )(SideMenu)
