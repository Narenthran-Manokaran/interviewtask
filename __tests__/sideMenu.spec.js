import React from 'react';
import { SideMenu, mapStateToProps, mapDispatchToProps } from './../src/component/sideMenu';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Header, Button, ListItem } from 'react-native-elements';

configure({ adapter: new Adapter() });

jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-navigation-stack', () => { return {Header: ()=>'whatever'} });

const initialState = {
  data: [],
  dataFetched: false,
  isFetching: false,
  error: false,
}

const items = [
  {
    params: {
      title: 'Test'
    }
  }
]

const navigation = {
  dispatch: () => {
    return true
  }
}

const fetchData = () => {
  return `${URI}/mostpopular/v2/viewed/1.json?${API_KEY}`;
}
const URI = 'https://api.nytimes.com/svc';
const API_KEY = 'api-key=okVdEHSsZSsT6Pluu48AAa35wmjr3BNj';

describe('SideMenu', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SideMenu appData={initialState} items={items} fetchData={fetchData}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should show previously rolled value', () => {
    expect(mapStateToProps({appData:initialState})).toEqual({appData:initialState});
  });
  it('should roll the dice again when button is clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).fetchData();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'FETCHING_DATA'});
  });
  test('test onPress functionality', () => {
    const onPressEvent = jest.fn();
    const wrapper = shallow(<SideMenu appData={initialState} items={items} fetchData={fetchData} onPress={ onPressEvent } navigation={navigation} />);
    wrapper.find(ListItem).first().props().onPress();
    expect(onPressEvent.mock.calls.length).toBe(0);
  });
})
