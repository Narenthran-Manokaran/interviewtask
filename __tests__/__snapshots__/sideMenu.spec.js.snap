// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "paddingTop": 20,
    }
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderColor": "#bcbbc1",
                "flexDirection": "row",
                "padding": 14,
              }
            }
          >
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": null,
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
            <View
              width={16}
            />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "fontSize": 17,
                  }
                }
                testID="listItemTitle"
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
              >
                Test
              </Text>
            </View>
            <View
              width={16}
            />
            <View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "justifyContent": "center",
                  }
                }
                theme={
                  Object {
                    "colors": Object {
                      "disabled": "hsl(208, 8%, 90%)",
                      "divider": "#bcbbc1",
                      "error": "#ff190c",
                      "grey0": "#393e42",
                      "grey1": "#43484d",
                      "grey2": "#5e6977",
                      "grey3": "#86939e",
                      "grey4": "#bdc6cf",
                      "grey5": "#e1e8ee",
                      "greyOutline": "#bbb",
                      "platform": Object {
                        "android": Object {
                          "error": "#f44336",
                          "primary": "#2196f3",
                          "secondary": "#9C27B0",
                          "success": "#4caf50",
                          "warning": "#ffeb3b",
                        },
                        "ios": Object {
                          "error": "#ff3b30",
                          "primary": "#007aff",
                          "secondary": "#5856d6",
                          "success": "#4cd964",
                          "warning": "#ffcc00",
                        },
                      },
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#8F0CE8",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  }
                }
                underlayColor="white"
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#517fa4",
                        "fontSize": 24,
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                  testID="iconIcon"
                >
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "lightgrey",
        "padding": 20,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2089dc",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Sign Out
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
